### PR TITLE
Fix build on FreeBSD

### PR DESCRIPTION
**Pull request by jemc**
(Changes by @mneumann with edits by @jemc to make it continue
to work on non-FreeBSD platforms as well)

---

This PR is a replacement for #208.

Closes #208.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -139,16 +139,20 @@ $(eval $(call MAKE_VAR_CACHE_FOR,RUNTIME_BITCODE_RELEASE_URL))
 LLVM_PATH?=$(BUILD)/llvm-static
 LLVM_CONFIG?=$(LLVM_PATH)/bin/llvm-config
 
+# Determine which flavor of the C++ standard library to link against.
+# We choose libstdc++ where available, or fallback to libc++.
+LIB_CXX_KIND?=$(shell ldconfig -p | grep -q libstdc++ && echo stdc++ || echo c++)
```
Comment (mneumann) [inline Makefile:144]: FreeBSD doesn't have option `-p` for `ldconfig` :/
I think we have to conditionalize it based on the platform.

**Review by mneumann — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (jemc) [reply Makefile]: Ah, that's unfortunate...

Okay, I force-pushed an amended commit that checks for `freebsd` in the platform string.

Comment (jemc) [reply Makefile]: Pushing another commit that also flips MacOS over to `libc++` in an attempt to fix the broken CI build from the last attempt.

**Comment by jemc** [comment]
@mneumann - please review and test on FreeBSD again, when you get the chance.

File: Makefile (hunk 1 above)
Comment (mneumann) [reply Makefile]: that works

**Comment by mneumann** [comment]
tested on FreeBSD. works like a charm

**Pull request merged**